docs/openapi.yml

```diff
@@ -1445,43 +1520,45 @@ components:
                 processPeak:
                   type: number
                   example: 0.090362
-    BasicInitialize:
+    UploadEnd:
+      type: object
+      required:
+        - timestamp
+        - fileList
+        - modifiedList
+      properties:
+        timestamp:
+          description: time since epoch of last upload
+          type: number
+          example: 157139320301
+        fileList:
+          description: an array of all files the project contains
+          type: array
+          example: ["/project/file1", "/project/file2", "project/modifiedfile1"]
+        modifiedList:
+          description: an array of files that have been modified since the last project upload
+          type: array
+          example: ["/project/modifiedfile1"]
+      
+    Upload:
       type: object
       required:
-        - status
-        - projectPath
+        - path
+        - msg
```

Comment:
We do have an optional `mode` parameter now so we can get the file permissions right as well as the path and the contents.